[Console] Migrate SCSS to Emotion

kapral18 opened the pull request:
Addresses https://github.com/elastic/kibana/issues/205027

• Migrate Console + Embeddable UI to emotion

### Screenshots (Before | After)

<img width="3201" height="1401" alt="image" src="https://github.com/user-attachments/assets/e48681f0-e356-4953-8884-6481bd59695e" />

<img width="3196" height="1398" alt="image" src="https://github.com/user-attachments/assets/846d90bd-ca38-4f1f-8a80-bf7e6c082d7a" />

<img width="3199" height="1385" alt="image" src="https://github.com/user-attachments/assets/b3a16f27-9dd5-4597-a913-6f285a0494da" />



<img width="3195" height="1398" alt="image" src="https://github.com/user-attachments/assets/86fe78c9-aba2-4c70-ad4e-e5e5d4fe7f92" />

<img width="3203" height="1405" alt="image" src="https://github.com/user-attachments/assets/ee2eb974-0809-467b-82f2-5d9b45c497d9" />

<img width="3204" height="1393" alt="image" src="https://github.com/user-attachments/assets/45b38019-6288-4170-a7b3-7b9fae41327f" />

<img width="3205" height="1397" alt="image" src="https://github.com/user-attachments/assets/e0af4f9e-46ae-4689-928a-85b50658fbfe" />

<img width="3199" height="1395" alt="image" src="https://github.com/user-attachments/assets/4a3cdb9b-3649-4637-ac57-e50e763d6dd8" />

kapral18 committed:
[Console] Migrate SCSS to Emotion

kapral18 committed:
chore: fix highlighted line background transparency

elasticmachine commented:
Pinging @elastic/kibana-management (Team:Kibana Management)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/343449)
* Commit: 05cef26b941cdac78622fadaa1e22fccf89bfee6

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/343449#01997d67-4505-4a12-a517-73c4b1a41268)

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343449","number":343449,"commit":"05cef26b941cdac78622fadaa1e22fccf89bfee6"}],"number":343449}
buildkite-pr-comment-->

kapral18 committed:
chore: fix ci types regression in monaco editor actions provider test

jovana-andjelkovic commented:
@kapral18 looks like the header color of embeddable changes to black in the after, not sure if that's intentional?

kapral18 committed:
chore: fix inverted background styles in embeddable console

jovana-andjelkovic commented:
@kapral18 looks good now, thanks!

ElenaStoeva commented on src/platform/plugins/shared/console/public/application/containers/config/config.tsx:
```diff
@@ -15,20 +15,30 @@ import {
+
 export function Config() {
   const isVerticalLayout = useIsWithinBreakpoints(['xs', 's', 'm']);
+  const styles = useStyles();
```
nit: the hook might be an overkill if we have a single css variable - wouldn't it be better to add just a const?

ElenaStoeva commented on src/platform/plugins/shared/console/public/application/containers/embeddable/embeddable_console.tsx:
```diff
@@ -86,7 +87,9 @@ export const EmbeddableConsole = ({
   );
-  const chromeStyle = useObservable(core.chrome.getChromeStyle$());
+  const isOpen = consoleState.view !== EmbeddableConsoleView.Closed;
+  const styles = useStyles({ chromeStyle, isOpen });
```
I wonder if it's worth memoizing the returned styles object to avoid rerendering of the child components, wdyt?

ElenaStoeva replied:
Oh actually I meant to memoize based on changes of `chromeStyle` and `isOpen`, sorry for not being clear enough. I was thinking that if for example the user opens the embeddable console and `isOpen` changes, this would trigger a new styles object, and so some of the child components that are not dependent on `isOpen` but use the styles would be rerendered unnecessarily, because the styles object is changed, no?

On a second thought, this probably doesn't make much difference as we don't expect `chromeStyle` and `isOpen` to change frequently, so I'd be fine to leave it as is.

ElenaStoeva replied:
I was actually thinking about options like wrapping the returned styles obejct in the hook with useMemo:

```
export const useStyles = ({
  chromeStyle,
  isOpen,
}: {
  chromeStyle?: ChromeStyle;
  isOpen: boolean;
}) => {
  const { euiTheme } = useEuiTheme();
  const variables = useEmbeddableConsoleStyleVariables();

  return useMemo(() => {
    return {
      embeddableConsoleGlobal: css`...`,
      embeddableConsole: css`...`,
      embeddableConsoleControls: css`...`,
      ...
    };
  }, [chromeStyle, isOpen, euiTheme, variables]);
};
```

or splitting dynamic vs static styles as there are styles that don’t depend on `chromeStyle` or `isOpen` and only `embeddableConsole` really depends on `chromeStyle` or `isOpen`.

But as I mentioned, `chromeStyle` and `isOpen` don't change that often and the current code is clean and easy to read, so think it's good as it is. 👍

kapral18 replied:
Yes that `useMemo()` would prevent re-serialization of `css()` blocks and recreation of the wrapper object which would be garbage collected. But even unmemoized those `css()` blocks won't cause a rerender downstream because of the css -> className compile time conversion and lookup from cache. Let discuss this in a call.  I will merge this for now.

ElenaStoeva commented on src/platform/plugins/shared/console/public/styles/index.ts:
```diff
@@ -0,0 +1,11 @@
+ */
+
+// This file exports styles that are used across the Console plugin.
+// Note: Prefer colocating styles with their usage when possible.
```
nit: Why do we keep the /styles directory now that it's empty?

ElenaStoeva reviewed (commented):
Great job @kapral18! Most of the code changes lgtm, left a few small suggestions.

kapral18 committed:
chore: addressing review feedback

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/344397)
* Commit: 68c849a71c3cb0144476013c2d9c58b855bda282
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-236377)

### Failed CI Steps
* [FTR Configs #6](https://buildkite.com/elastic/kibana-pull-request/builds/344397#0199867b-43ae-4376-81b0-4a20b142ecb6)
* [FTR Configs #6](https://buildkite.com/elastic/kibana-pull-request/builds/344397#019986a7-f798-443f-851c-c0b1041ccad9)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/344397#0199867b-43ae-4376-81b0-4a20b142ecb6) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/344397/jobs/0199867b-43ae-4376-81b0-4a20b142ecb6/artifacts/019986a7-de74-4add-8b39-ca2b65c76321) FTR Configs #<span></span>6 / Serverless Common UI - Dev Tools Search Profiler Editor With a test index profiles a simple query
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/344397#019986a7-f798-443f-851c-c0b1041ccad9) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/344397/jobs/019986a7-f798-443f-851c-c0b1041ccad9/artifacts/019986be-e82f-4838-95d2-96a3b66ac3e9) FTR Configs #<span></span>6 / Serverless Common UI - Dev Tools Search Profiler Editor With a test index profiles a simple query

### History
* :yellow_heart: [Build #344183](https://buildkite.com/elastic/kibana-pull-request/builds/344183) was flaky 6ad9781406d896cf69de75968835fc904fca773c
* :broken_heart: [Build #343981](https://buildkite.com/elastic/kibana-pull-request/builds/343981) failed f049259e0c4774dfdc92c52b8219ed360068dc2c
* :broken_heart: [Build #343449](https://buildkite.com/elastic/kibana-pull-request/builds/343449) failed 05cef26b941cdac78622fadaa1e22fccf89bfee6

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/344397","number":344397,"commit":"68c849a71c3cb0144476013c2d9c58b855bda282"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/344183","number":344183,"commit":"6ad9781406d896cf69de75968835fc904fca773c"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343981","number":343981,"commit":"f049259e0c4774dfdc92c52b8219ed360068dc2c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343449","number":343449,"commit":"05cef26b941cdac78622fadaa1e22fccf89bfee6"}],"number":344397}
buildkite-pr-comment-->

kapral18 committed:
chore: fix background color to use backgroundBaseSubdued

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/344513)
* Commit: 17c9edefdf05c9922e7db8d6850bcb00c805cac6
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-236377)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/b62d4ffaad573b6f64ec216eef6f77222766625b) | [after](https://github.com/elastic/kibana/commit/17c9edefdf05c9922e7db8d6850bcb00c805cac6) | diff |
| --- | --- | --- | --- |
| `console` | 247 | 233 | -14 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/b62d4ffaad573b6f64ec216eef6f77222766625b) | [after](https://github.com/elastic/kibana/commit/17c9edefdf05c9922e7db8d6850bcb00c805cac6) | diff |
| --- | --- | --- | --- |
| `console` | 195.3KB | 181.6KB | -13.6KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/b62d4ffaad573b6f64ec216eef6f77222766625b) | [after](https://github.com/elastic/kibana/commit/17c9edefdf05c9922e7db8d6850bcb00c805cac6) | diff |
| --- | --- | --- | --- |
| `console` | 28.0KB | 27.9KB | -76.0B |



### History
* :broken_heart: [Build #344397](https://buildkite.com/elastic/kibana-pull-request/builds/344397) failed 68c849a71c3cb0144476013c2d9c58b855bda282
* :yellow_heart: [Build #344183](https://buildkite.com/elastic/kibana-pull-request/builds/344183) was flaky 6ad9781406d896cf69de75968835fc904fca773c
* :broken_heart: [Build #343981](https://buildkite.com/elastic/kibana-pull-request/builds/343981) failed f049259e0c4774dfdc92c52b8219ed360068dc2c
* :broken_heart: [Build #343449](https://buildkite.com/elastic/kibana-pull-request/builds/343449) failed 05cef26b941cdac78622fadaa1e22fccf89bfee6

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/344513","number":344513,"commit":"17c9edefdf05c9922e7db8d6850bcb00c805cac6"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/344397","number":344397,"commit":"68c849a71c3cb0144476013c2d9c58b855bda282"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/344183","number":344183,"commit":"6ad9781406d896cf69de75968835fc904fca773c"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343981","number":343981,"commit":"f049259e0c4774dfdc92c52b8219ed360068dc2c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343449","number":343449,"commit":"05cef26b941cdac78622fadaa1e22fccf89bfee6"}],"number":344513}
buildkite-pr-comment-->

ElenaStoeva reviewed (approved):
Thanks for fixing the issue from the missing semicolon and the colors! Looks good now!

kapral18 committed:
chore: address remaining comment and change embeddableConsole styles

- addressed leftover EuiThemeContext
- extracted conditionals out of useStyles to array of css props
aligning with how it was originally done with classnames

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/345356)
* Commit: 79098542fd08b375de2295073ac044b4ea5f0e69
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-236377)

### Failed CI Steps
* [FTR Configs #52](https://buildkite.com/elastic/kibana-pull-request/builds/345356#0199970b-f041-4808-b0b7-1d8bc0e1a45d)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/345356#0199970b-f041-4808-b0b7-1d8bc0e1a45d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/345356/jobs/0199970b-f041-4808-b0b7-1d8bc0e1a45d/artifacts/01999734-2e3b-4258-997b-9d888b952383) FTR Configs #<span></span>52 / integrations When on the Trusted Apps list "after all" hook for "should be able to add a new trusted app and remove it"

The CI Stats report is too large to be displayed here, check out the CI build annotation for this information.

### History
* :broken_heart: [Build #345222](https://buildkite.com/elastic/kibana-pull-request/builds/345222) failed 81e72af2fd43fe28fb04de79529ec808d711a18a
* :green_heart: [Build #344513](https://buildkite.com/elastic/kibana-pull-request/builds/344513) succeeded 17c9edefdf05c9922e7db8d6850bcb00c805cac6
* :broken_heart: [Build #344397](https://buildkite.com/elastic/kibana-pull-request/builds/344397) failed 68c849a71c3cb0144476013c2d9c58b855bda282
* :yellow_heart: [Build #344183](https://buildkite.com/elastic/kibana-pull-request/builds/344183) was flaky 6ad9781406d896cf69de75968835fc904fca773c
* :broken_heart: [Build #343981](https://buildkite.com/elastic/kibana-pull-request/builds/343981) failed f049259e0c4774dfdc92c52b8219ed360068dc2c
* :broken_heart: [Build #343449](https://buildkite.com/elastic/kibana-pull-request/builds/343449) failed 05cef26b941cdac78622fadaa1e22fccf89bfee6

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/345356","number":345356,"commit":"79098542fd08b375de2295073ac044b4ea5f0e69"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/345222","number":345222,"commit":"81e72af2fd43fe28fb04de79529ec808d711a18a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/344513","number":344513,"commit":"17c9edefdf05c9922e7db8d6850bcb00c805cac6"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/344397","number":344397,"commit":"68c849a71c3cb0144476013c2d9c58b855bda282"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/344183","number":344183,"commit":"6ad9781406d896cf69de75968835fc904fca773c"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343981","number":343981,"commit":"f049259e0c4774dfdc92c52b8219ed360068dc2c"}],"number":345356}
buildkite-pr-comment-->
